src/pathfinding_service/service.py

```diff
@@ -138,6 +149,15 @@ def follows_token_network(self, token_network_address: TokenNetworkAddress) -> b
         """ Checks if a token network is followed by the pathfinding service. """
         return token_network_address in self.token_networks.keys()
 
+    def handle_reachability_change(
+        self, address: Address, reachability: AddressReachability
+    ) -> None:
+        log.debug("Updating reachability", address=address, status=reachability)
+
+        for token_network_model in self.token_networks.values():
+            if address in token_network_model.G.nodes:
+                token_network_model.address_to_reachability[address] = reachability
```

Comment:
Iterating through all nodes of all token networks is not really nice. Can't we do
```python
try:
    token_network_model.address_to_reachability[address] = reachability
except KeyError:
    pass
```
?

Reply:
Wouldn't the above code assign the reachability for any case and never throw a `KeyError`?

Reply:
Right, make that
```
if address in token_network_model.address_to_reachability:
    token_network_model.address_to_reachability[address]
```
and set it to unknown when the node is first seen.

Reply:
Then we iterate over the keys of `address_to_reachability` instead of `G.nodes`. Does that improve things?

I'd say the other way is to store presence in the service, not in the token network and use that global map during routing.

Reply:
One is a dict, the other is a list, so yes. But we can leave this for later. We'll probably move things around a few times, anyway.